zebra-network/src/peer/handshake.rs

```diff
@@ -180,6 +180,138 @@ where
     }
 }
 
+/// Negotiate the Zcash network protocol version with the remote peer
+/// at `addr`, using the connection `peer_conn`.
+///
+/// We split `Handshake` into its components before calling this function,
+/// to avoid infectious `Sync` bounds on the returned future.
```

Comment:
'infectious' is a good way to put this 😄

Reply:
I tried it, and they just got everywhere